tachyon/math/base/semigroups.h

```diff
@@ -197,25 +198,7 @@ class MultiplicativeSemigroup {
     return g;
   }
 
-  // generator: g
-  // return: [c, c * g, c * g², ..., c * g^{|size| - 1}]
-  constexpr static std::vector<MulResult> GetSuccessivePowers(
-      size_t size, const G& generator, const G& c = G::One()) {
-    std::vector<MulResult> ret(size);
-    base::Parallelize(
-        ret, [&generator, &c](absl::Span<G> chunk, size_t chunk_offset,
-                              size_t chunk_size) {
-          MulResult pow = generator.Pow(chunk_offset * chunk_size);
-          if (!c.IsOne()) pow *= c;
-          for (G& v : chunk) {
-            v = pow;
-            pow *= generator;
-          }
-        });
-    return ret;
-  }
-
-  constexpr static std::pmr::vector<MulResult> GetSuccessivePowersPmr(
```

Comment:
Can you get rid of this intermediate change?

Reply:
I think that would be not so simple. Changing the `GetSuccessivePowers` to `std::pmr::vector` by default would cause the `UnivariateEvaluations` to use pmr vector.